packages/atproto_client/models/string_formats.py

```diff
@@ -193,13 +206,7 @@ def validate_nsid(v: str, _: ValidationInfo) -> str:
     Raises:
         ValueError: If NSID format is invalid
     """
-    if (
-        not atproto_core_validate_nsid(v, soft_fail=True)
-        or len(v) > MAX_NSID_LENGTH
-        or any(c in v for c in '@_*#!')  # Explicitly disallow special chars
-        or any(len(seg) > 63 for seg in v.split('.'))  # Max segment length
-        or any(seg[-1].isdigit() for seg in v.split('.'))  # No segments ending in numbers
-    ):
+    if not atproto_core_validate_nsid(v, soft_fail=True):
         raise ValueError(
             f'Invalid NSID: must be dot-separated segments (e.g. app.bsky.feed.post) with max length {MAX_NSID_LENGTH}'
         )
```

Comment:
```suggestion
    try:
        atproto_core_validate_nsid(v)
        return v
    except InvalidNsidError as e:
        raise ValueError(f'Invalid NSID: {e!s}') from None
```

let's proxy error messages here; and delete `MAX_NSID_LENGTH` const

Reply:
ah, good call! updated in [22aff21](https://github.com/MarshalX/atproto/pull/572/commits/22aff2117575fd35560f3bdcc626baf872e6163d)